test/soak/serve_hostnames/serve_hostnames.go

```diff
@@ -86,31 +93,45 @@ func main() {
 	glog.Infof("Using namespace %s for this test.", ns)
 
 	// Create a service for these pods.
-	glog.Info("Creating service serve-hostnames")
-	svc, err := c.Services(ns).Create(&api.Service{
-		ObjectMeta: api.ObjectMeta{
-			Name: "serve-hostnames",
-			Labels: map[string]string{
-				"name": "serve-hostname",
+	glog.Infof("Creating service %s/serve-hostnames", ns)
+	// Make several attempts to create a service.
+	var svc *api.Service
+	for start := time.Now(); time.Since(start) < serviceCreateTimeout; time.Sleep(2 * time.Second) {
+		t := time.Now()
+		svc, err = c.Services(ns).Create(&api.Service{
+			ObjectMeta: api.ObjectMeta{
+				Name: "serve-hostnames",
+				Labels: map[string]string{
+					"name": "serve-hostname",
+				},
 			},
-		},
-		Spec: api.ServiceSpec{
-			Port:       9376,
-			TargetPort: util.NewIntOrStringFromInt(9376),
-			Selector: map[string]string{
-				"name": "serve-hostname",
+			Spec: api.ServiceSpec{
+				Port:       9376,
+				TargetPort: util.NewIntOrStringFromInt(9376),
+				Selector: map[string]string{
+					"name": "serve-hostname",
+				},
 			},
-		},
-	})
+		})
+		glog.Infof("Service create %s/server-hostnames took %v", ns, time.Since(t))
```

Comment:
I want to print the failed attempts. In the next iteration I plan to add a `--verbose` flag to make the default output quieter.